### PR TITLE
feat: add exercise E The Odin Project

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md [hunk 1]
```diff
@@ -7,38 +7,165 @@ dashedName: css-foundations-exercise-e
...
+    for(let i = 0; i < styles.length; i++){
+        if(styles[i] != undefined){
+            return styles[i];
+        }
+    }
+}
+
+assert(getCorrectStyle()?.color === 'red');
+
+console.log(getCorrectStyle()?.color)
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md:65]:
```suggestion
```

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md [hunk 2]
```diff
@@ -7,38 +7,165 @@ dashedName: css-foundations-exercise-e
...
 
+The goal of this exercise is to apply styles to elements that are descendants of another element, while leaving elements that aren't descendants of that element unstyled.
+
+1. You should see a `yellow` background, `p` elements that are descendants of the `div` element.
+1. You should see a text color of `red` elements that are descendants of the `div` element.
+1. You should see a font size of `20px` elements that are descendants of the `div` element.
+1. You should text that is center aligned for elements that are descendants of the `div` element.
+
+
+red text, a font size of 20px, and center aligned.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md:20]:
```suggestion
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Review by naomi-lgbt — APPROVED** [review]
I wanted to DRY this out with the `--after-seed-code--` but that triggers the JavaScript editor to appear 😔 

This is good as is.

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md [hunk 3]
```diff
@@ -7,38 +7,161 @@ dashedName: css-foundations-exercise-e
...
+
+1. You should see a `yellow` background, `p` elements that are descendants of the `div` element.
+1. You should see a text color of `red` elements that are descendants of the `div` element.
+1. You should see a font size of `20px` elements that are descendants of the `div` element.
+1. You should text that is center aligned for elements that are descendants of the `div` element.
 
 # --hints--
 
-Test 1
+You should have a background color of `yellow` on your decendants.
```
Comment (Nirajn2311) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md:21]:
```suggestion
You should have a background color of `yellow` on your descendants.
```

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md [hunk 4]
```diff
@@ -7,38 +7,161 @@ dashedName: css-foundations-exercise-e
...
+        }
+    }
+}
+
+assert(getCorrectStyle()?.backgroundColor === 'yellow');
 
 ```
 
-Test 2
+You should have a text color of `red` on your decendants.
```
Comment (Nirajn2311) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md:43]:
```suggestion
You should have a text color of `red` on your descendants.
```

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md [hunk 5]
```diff
@@ -7,38 +7,161 @@ dashedName: css-foundations-exercise-e
...
+        }
+    }
+}
+
+assert(getCorrectStyle()?.color === 'red');
 
 ```
 
-Test 3
+You should have a font size of `20px` on your decendants.
```
Comment (Nirajn2311) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md:65]:
```suggestion
You should have a font size of `20px` on your descendants.
```

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md [hunk 6]
```diff
@@ -7,38 +7,161 @@ dashedName: css-foundations-exercise-e
...
+            return styles[i];
+        }
+    }
+}
+
+assert(getCorrectStyle()?.fontSize === '20px');
 
 ```
 
+You should center align the text on your decendants.
```
Comment (Nirajn2311) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md:87]:
```suggestion
You should center align the text on your descendants.
```

File: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md [hunk 7]
```diff
@@ -7,38 +7,161 @@ dashedName: css-foundations-exercise-e
...
 
-description
+Understanding how combinators work can become a lot easier when you start playing around with them and see what exactly is affected by them versus what isn't.
+
+The goal of this exercise is to apply styles to elements that are descendants of another element, while leaving elements that aren't descendants of that element unstyled.
+
+1. You should see a `yellow` background, `p` elements that are descendants of the `div` element.
+1. You should see a text color of `red` elements that are descendants of the `div` element.
+1. You should see a font size of `20px` elements that are descendants of the `div` element.
+1. You should text that is center aligned for elements that are descendants of the `div` element.
```
Comment (Nirajn2311) [inline curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md:17]:
```suggestion
1. You should see a `yellow` background for `p` elements that are descendants of the `div` element.
1. You should see a text color of `red` for elements that are descendants of the `div` element.
1. You should see a font size of `20px` for elements that are descendants of the `div` element.
1. You should center align text for elements that are descendants of the `div` element.
```

**Review by Nirajn2311 — CHANGES_REQUESTED** [review]
Grammatical and spelling errors. Rest all are good to go.

**Review by Nirajn2311 — APPROVED** [review]

**Pull request merged**